wifiphisher/common/interfaces.py

```diff
@@ -517,6 +551,9 @@ def get_interface(self, has_ap_mode=None, has_monitor_mode=None):
 
         # return interface if found otherwise raise an error
         if chosen_interface:
+            adapter = self._name_to_object[chosen_interface]
+            if adapter.is_managed_by_nm:
```

Comment:
As I was fixing the test cases I noticed that placing this will not be optimal. We should incorporate this condition to the algorithm. As an example imagine user has the following interfaces:
- wlan0 AP MON --> not managed
- wlan1 AP MON --> managed
- wlan2 AP MON --> not managed
The current algorithm fails on this case.